lake/api/api.go

```diff
@@ -22,8 +22,8 @@ import (
 
 type Interface interface {
 	Root() *lake.Root
-	Query(ctx context.Context, head *lakeparse.Commitish, src string, srcfiles ...string) (zio.ReadCloser, error)
-	QueryWithControl(ctx context.Context, head *lakeparse.Commitish, src string, srcfiles ...string) (zbuf.ProgressReadCloser, error)
+	Query(ctx context.Context, head *lakeparse.Commitish, src string) (zio.ReadCloser, error)
+	QueryWithControl(ctx context.Context, head *lakeparse.Commitish, src string) (zbuf.ProgressReadCloser, error)
```

Comment:
Do these APIs need to change?

Reply:
I think it should be this way since if you let `lake/api/api` do the ConcatSources for you won't be able to map error locations to your source set.

Reply:
I was thinking that the mapping would continue to happen inside these methods in the same way it effectively does now. Shifting that burden to the caller seems like it makes for a more complicated interface since the caller now has to deal with parser.ConcatSource and clierrors.Format.

Does shifting that burden have some benefit I'm not seeing?

Reply:
I eventually want to rewrite the error text to use asni escape sequences for pretty console output- where should the package that formats the error messages go?